### DISCUSSION TITLE
Introducing expressions in Postman Flow

### DISCUSSION THREAD

**Discussion by samay-v** [postmanlabs/postman-flows#124] [Announcements]
Data is a first-class citizen in Flow, to extract the most value from this data we felt the need to introduce a way to operate and transform that data that is: 

1. **Consistent:** Can be reused in more than one block
2. **Powerful:** Can be used to do complex data manipulations
3. **Learnable:** Should be easy to become efficient at.

## The expression input

![2021-11-15_12-16-21 (1)](https://user-images.githubusercontent.com/28357106/141736251-7cb58ad5-e177-4525-95d5-efa235c72012.gif)

The idea is to have one input field that takes in an expression with any data type and operate them to get any kind of value. 

> Here data refers to references to input received by the block as well as data entered by the user in the input

The expression input will provide the ability to refer to the input the block as received, enter a value of any data type and various operators to operate them with. 

With the new primitive data types established, the expression input can be slightly more helpful:

1. It can predict the type of input you're entering. 
2. It can suggest the operators based on the data type of the operand
3. It can validate if the expression resolves to the correct data type.
4. It also pushes operators on top that take you closer to the required resolution of the input.

This input will be available as a part of a new "Check" block from today. We intend to extend it and use it in all blocks that need data as configuration.

**Comment by sameershaik1996** [comment]
Is this feature live?

**Reply by anudeepmedicharla** [reply]
I think it went live today. @samay-v please correct me if I'm wrong :)

**Comment by csantelman** [comment]
Is there anyway to do the 'contains' operator?  Like you could in the depricated 'Validate' block?  

I want to look to see if one of the json field's data contains a particular sub-string.

**Reply by saswatds** [reply]
@csantelman Can you take a look if any of the operators here would fit your need https://www.postmanlabs.com/postman-flows/docs/operators/ ? I think you are looking for the `in` operator here. If not then let me know of your use-case we will evaluate and add a new operator

**Comment by csantelman** [comment]
The condition "in" operator does not work the same was as the old validate "contains" operator.

Let's say my api returns the following

[
{
"featureName": "find",
"createdBy": null
},
{
"featureName": "miss",
"createdBy": null
},
{
"featureName": "find.this",
"createdBy": null
},
{
"featureName": "miss.this",
"createdBy": null
},
{
"featureName": "please.find.this",
"createdBy": null
},
]

I want to grab all the rows that contain a feature name that has 'find' anywhere in the field.

If I use the Condition operator with the following rule

/data/featureName in "find" 

It will only find the row that matches exactly.

But the old Validate operator with the rule
$ Record > featureName
Operator : contains
Value : find

Will find the 3 rows as expected.

Hope this helps

Craig

**Reply by dchaitu** [reply]
Then if you want to search if null is present or value is null. How to do that?
